src/components/app/DocumentList/index.js

```diff
@@ -0,0 +1,855 @@
+          const parsed = parseToDisplay(row.fieldsByName);
+          resultById[`${row.id}`] = parsed;
+          row.fieldsByName = parsed;
+        });
+
+        const pageColumnInfosByFieldName = response.columnsByFieldName;
+
+        mergeColumnInfosIntoViewRows(
+          pageColumnInfosByFieldName,
+          response.result
+        );
+
+        if (this.mounted) {
+          const newState = {
+            pageColumnInfosByFieldName: pageColumnInfosByFieldName,
+            triggerSpinner: false,
+          };
+
+          if (response.filters) {
+            newState.filtersActive = filtersToMap(response.filters);
+          }
+
+          if (
+            locationAreaSearch ||
+            (newState.filtersActive &&
+              newState.filtersActive.has(`location-area-search`))
+          ) {
+            this.getLocationData(resultById);
+          }
+
+          this.setState({ ...newState }, () => {
+            if (forceSelection && response && result && result.length > 0) {
+              const selection = [result[0].id];
+
+              selectTableItems({
+                windowType,
+                viewId,
+                ids: selection,
+              });
+            }
+          });
+
+          // process modal specific
+          const { parentViewId, parentWindowId, headerProperties } = response;
+
+          updateRawModal(windowType, {
+            parentViewId,
+            parentWindowId,
+            headerProperties,
+          });
+        }
+
+        indicatorState('saved');
+      })
+      .catch(() => {
+        this.setState({ triggerSpinner: false });
+      });
+  };
+
+  getLocationData = (resultById) => {
+    const {
+      windowType,
+      viewId,
+      reduxData: { mapConfig },
+    } = this.props;
+
+    locationSearchRequest({ windowId: windowType, viewId }).then(({ data }) => {
+      const locationData = data.locations.map((location) => {
+        const name = get(
+          resultById,
+          [location.rowId, 'C_BPartner_ID', 'value', 'caption'],
+          location.rowId
+        );
+
+        return {
+          ...location,
+          name,
+        };
+      });
+
+      if (locationData.length) {
+        updateViewData(locationData, windowType);
+      }
+
+      if (mapConfig && mapConfig.provider) {
+        // for mobile show map
+        // for desktop show half-n-half
+        this.setState({ panelsState: GEO_PANEL_STATES[1] });
+      }
+    });
+  };
+
+  // MANAGING SORT, PAGINATION, FILTERS --------------------------------------
+
+  /**
+   * @method handleChangePage
+   * @summary ToDo: Describe the method.
+   */
+  handleChangePage = (index) => {
+    const { reduxData } = this.props;
+    let currentPage = reduxData.page;
+
+    switch (index) {
+      case 'up':
+        currentPage * reduxData.pageLength < reduxData.size
+          ? currentPage++
+          : null;
+        break;
+      case 'down':
+        currentPage != 1 ? currentPage-- : null;
```

Comment:
!==

Reply:
Yeah, let's hope somebody somewhere didn't decide on using a string here ;)

Reply:
I'm just pointing this out to have a better code quality.